google-cloud-bigquery/lib/google/cloud/bigquery/schema.rb

```diff
@@ -276,141 +244,86 @@ def date name, description: nil, mode: :nullable
         #   end
         #
         def record name, description: nil, mode: nil
+          # TODO: do we need to fail if no block was given?
```

Comment:
Right, the BigQuery API will enforce that a record field has nested fields and return an appropriate error if they do not. But passing a block is no guarantee that a field will be added.

```ruby
require "google/cloud/bigquery"

bigquery = Google::Cloud::Bigquery.new
dataset = bigquery.dataset "my_dataset"

table = dataset.create_table "things" do |schema|
  schema.record "properties" do |properties|
    # not doing anything here but passing a block
    # no nested records being added
  end
end
```

I don't think its needed anymore, but this is something I'd like to discuss with @quartzmo when he gets back. If we did remove the error, users _could_ decide to use code like this:

```ruby
require "google/cloud/bigquery"

bigquery = Google::Cloud::Bigquery.new
dataset = bigquery.dataset "my_dataset"

table = dataset.create_table "things" do |schema|
  properties = schema.record "properties"
  properties.string "name"
  properties.integer "size"
end
```